### PR TITLE
Axios: added retry count and allow shorthand interval in config

### PR DESCRIPTION
**Pull request by yaswanth-godspeed**
added retry count and allow shorthand interval in config

### CODE REVIEW

File: plugins/axios-as-datasource/src/index.ts [hunk 1]
```diff
@@ -21,6 +23,89 @@ class DataSource extends GSDataSource {
...
+      
+              case 'exponential':
+                const delay = 2 ** retryNumber * interval;
+                const randomSum = delay * 0.2 * Math.random();
+                logger.info(`Attempt ${retryNumber}: Retrying request with ${type} retry delay. Error: ${error.message}`);
+                return delay + randomSum;
+            }
+            return 0;
+          },
+          retryCondition: (error) => {
```
Comment (siddarthvader) [inline plugins/axios-as-datasource/src/index.ts:55]: does this have to be a function or can we just set it to 
`retryCondition:true`

File: plugins/axios-as-datasource/src/index.ts [hunk 2]
```diff
@@ -21,6 +23,89 @@ class DataSource extends GSDataSource {
...
+        
+                case 'exponential':
+                  const delay = 2 ** retryNumber * interval;
+                  const randomSum = delay * 0.2 * Math.random();
+                  logger.info(`Attempt ${retryNumber}: Retrying request with ${type} retry delay. Error: ${error.message}`);
+                  return delay + randomSum;
+              }
+              return 0;
+            },
+            retryCondition: (error) => {
```
Comment (siddarthvader) [inline plugins/axios-as-datasource/src/index.ts:102]: same as above

**Review by siddarthvader — CHANGES_REQUESTED** [review]
@yaswanth-godspeed also write what are the default values of min interval and stuff, please add readme file and update the readme file of plugins root directory also.

File: plugins/axios-as-datasource/src/index.ts (hunk 1 above)
Comment (yaswanth-godspeed) [reply plugins/axios-as-datasource/src/index.ts:55]: retryCondition is a callback to further control if a request should be retried. default: it retries if it is a network error or a 5xx error on an idempotent request (GET, HEAD, OPTIONS, PUT or DELETE).

Setting retryCondition: true means that the request will always be retried, regardless of the error. If you want more control over when to retry requests, you should customize the retryCondition function based on the error received. For example, you might want to retry only on certain status codes or under specific error conditions. If you don't have specific conditions to check and want to retry on any error, you can leave it as is (return true).

**Pull request closed without merging**